pyk/src/pyk/cterm.py

```diff
@@ -31,15 +31,28 @@ def cterm(self) -> KInner:
     def hash(self) -> str:
         return self.cterm.hash
 
-    def match(self, pattern: 'CTerm') -> Optional[Tuple[Subst, KInner]]:
+    def match(self, pattern: 'CTerm') -> Optional[Subst]:
+        match_res = self.match_with_condition(pattern)
+
+        if not match_res:
+            return None
+
+        subst, condition = match_res
+
+        if condition != TOP:
+            return None
+
+        return subst
+
+    def match_with_condition(self, pattern: 'CTerm') -> Optional[Tuple[Subst, KInner]]:
         subst = match(pattern=pattern.config, term=self.config)
 
         if subst is None:
             return None
 
-        obligation = self._ml_impl(self.constraints, map(subst, pattern.constraints))
+        condition = self._ml_impl(self.constraints, map(subst, pattern.constraints))
 
-        return subst, obligation
+        return subst, condition
```

Comment:
I would call this `match_with_constraint` instead, to match the notion of `constraint` we're using elsewhere. Specifically, it's a condition which can be any number of clauses, but it's _intended_ to be attached to exactly 1 configuration clause.